src/store/index.ts

```diff
@@ -59,19 +60,25 @@ export const getPersistedState = () => {
   return getPreloadedState(persistedSlices)
 }
 
-const hydrationReducer: typeof rootReducer = (state, action) => {
+export const _hydrationReducer: typeof rootReducer = (state, action) => {
   if (action.type === HYDRATE_ACTION) {
-    return {
-      ...state,
-      ...action.payload,
-    }
+    /**
+     * When changing the schema of a Redux slice, previously stored data in LS might become incompatible.
+     * To avoid this, we should always migrate the data on a case-by-case basis in the corresponding slice.
+     * However, as a catch-all measure, we attempt to merge the stored data with the initial Redux state,
+     * so that any new properties added to the slice are not overwritten as undefined.
```

Comment:
```suggestion
     * so that any newly added properties in the initial state are preserved, and existing properties are taken from the LS.
```

A small edit.